.gitlab/release.yaml

```diff
@@ -0,0 +1,120 @@
+  stage: changelog
+  variables:
+    GIT_DEPTH: 0  # Always get the full history
+    GIT_STRATEGY: clone  # Always get the full history
+  tags:
+    - hetzner-amd-beefy
+  rules:
+    - if: $CI_COMMIT_BRANCH =~ "/^v\d+\.\d+\.x$/"
+    - if: $CI_COMMIT_BRANCH == $DEFAULT_BRANCH
+    # Exclude release tags
+    - if: $CI_COMMIT_TAG
+      when: never
+  before_script:
+    # Setting up git
+    - git config --global user.email "${GITHUB_USER_EMAIL}"
+    - git config --global user.name "${GITHUB_USER_NAME}"
+    # GITHUB_TOKEN for Github cli authentication
+    - export GITHUB_TOKEN=${GITHUB_CLI_TOKEN}
+    # getting the centralized git cliff config
+  script:
+    - release-please release-pr
+      --token=${GITHUB_BOT_TOKEN_REPO_FULL}
+      --repo-url=${GITHUB_REPO_URL}
+      --target-branch=${CI_COMMIT_REF_NAME} || echo "INFO - release already exists" # workaround because we shifted to prerelease versioning strategy and there's already a PR open
+    # git cliff: override the changelog
+    - test $GIT_CLIFF == "false" && echo "INFO - Skipping git-cliff" && exit 0
+    - git remote add github-${CI_JOB_ID} https://${GITHUB_USER_NAME}:${GITHUB_BOT_TOKEN_REPO_FULL}@github.com/${GITHUB_REPO_URL} || true  # Ignore already existing remote
+    - gh repo set-default https://${GITHUB_USER_NAME}:${GITHUB_BOT_TOKEN_REPO_FULL}@github.com/${GITHUB_REPO_URL}
+    - RELEASE_PLEASE_PR=$(gh pr list --author "${GITHUB_USER_NAME}" --head "release-please--branches--${CI_COMMIT_REF_NAME}" --json number | jq -r '.[0].number // empty')
+    - test -z "$RELEASE_PLEASE_PR" && echo "No release-please PR found" && exit 0
+    - cp CHANGELOG.md CHANGELOG.md.${CI_COMMIT_SHA}
+    - gh pr checkout --force $RELEASE_PLEASE_PR
+    - mv CHANGELOG.md.${CI_COMMIT_SHA} CHANGELOG.md
+    - wget --output-document cliff.toml https://raw.githubusercontent.com/mendersoftware/mendertesting/master/utils/cliff.toml
+    - RELEASE_VERSION="$(jq -r '.["."]' .release-please-manifest.json)"
+    - git cliff --unreleased --prepend CHANGELOG.md --github-repo ${GITHUB_REPO_URL} --use-branch-tags --tag ${RELEASE_VERSION}
+    - git add CHANGELOG.md
+    - git commit --amend -s --no-edit
+    - git push github-${CI_JOB_ID} --force
+    # Update the PR body
+    - git cliff --unreleased --bump -o tmp_pr_body.md --github-repo ${GITHUB_REPO_URL} --use-branch-tags
+    - gh pr edit $RELEASE_PLEASE_PR --body-file tmp_pr_body.md
+    - rm tmp_pr_body.md
+  after_script:
+    - git remote remove github-${CI_JOB_ID}
+
+release:github:
+  image: "registry.gitlab.com/northern.tech/mender/mender-test-containers:release-please-v1-master"
+  stage: .post
+  tags:
+    - hetzner-amd-beefy
+  rules:
+    # Only make available for protected branches (main and maintenance branches)
+    - if: '$CI_COMMIT_TAG =~ /^v\d+\.\d+\.\d+$/'
+      when: never
+    - if: $CI_COMMIT_BRANCH =~ "/^v\d+\.\d+\.x$/" && $CI_COMMIT_BRANCH == $DEFAULT_BRANCH
+      when: manual
```

Comment:
```
      when: manual
```
Actually I'm a bit uncertain about this: you have to consider what's your release trigger: right now in mender-server and app-update-modules we have two release triggers: 1. merge down the CHANGELOG pr and 2. run this `release:github:` manual job; but with mender-helm this last job is automatic so the only release trigger is to merge down the CHANGELOG PR.

It depends on how you feel comfortable with the release triggers; I'm personally more keen to simplify this process and have just a single release trigger: merging down the CHANGELOG PR. But your call here.

Reply:
I prepared the release process based on the mender-server one (see [JIRA template here](https://northerntech.atlassian.net/browse/QA-979)) so with that process would be two steps: merge down the changelog PR and then manually run release:github. I think I can keep that at least for this repository that might require manual tests. For the independent tools (the famous 3C release process) we could do the same as for mender-helm and do one step only.